File: en/java-spring-webflux/src/main/java/com/example/controller/OrderController.java
package com.example.controller;

import org.springframework.web.bind.annotation.*;
import com.example.model.Order;
import com.example.dto.*;
import jakarta.validation.Valid;
import java.util.List;
import java.math.BigDecimal;
import java.util.List;

@RestController
@RequestMapping("/api/orders")
public class OrderController {
    
    @GetMapping
    public Result<PageResult<Order>> listOrders(
            @RequestParam(required = false) Long userId,
            @RequestParam(required = false) String orderNo,
            @RequestParam(required = false) Integer status,
            @RequestParam(required = false) String startDate,
            @RequestParam(required = false) String endDate,
            @RequestParam(defaultValue = "1") Integer pageNum,
            @RequestParam(defaultValue = "10") Integer pageSize) {
        return Result.success(new PageResult<>());
    }
    
    @PostMapping
    public Result<Order> createOrder(@Valid @RequestBody CreateOrderReq req) {
        return Result.success(new Order());
    }
    
    @GetMapping("/{id}")
    public Result<Order> getOrder(@PathVariable Long id) {
        return Result.success(new Order());
    }
    
    @PutMapping("/{id}")
    public Result<Order> updateOrder(@PathVariable Long id, @Valid @RequestBody UpdateOrderReq req) {
        return Result.success(new Order());
    }
    
    @DeleteMapping("/{id}")
    public Result<Void> deleteOrder(@PathVariable Long id) {
        return Result.success(null);
    }
    
    @PostMapping("/{id}/cancel")
    public Result<Order> cancelOrder(@PathVariable Long id, @RequestParam(required = false) String reason) {
        return Result.success(new Order());
    }
    
    @PutMapping("/{id}/status")
    public Result<Order> updateStatus(@PathVariable Long id, @RequestParam Integer status) {
        return Result.success(new Order());
    }
    
    @GetMapping("/{id}/track")
    public Result<OrderTrackVO> trackOrder(@PathVariable Long id) {
        return Result.success(new OrderTrackVO());
    }
    
    @GetMapping("/user/{userId}")
    public Result<PageResult<Order>> getUserOrders(
            @PathVariable Long userId,
            @RequestParam(required = false) Integer status,
            @RequestParam(defaultValue = "1") Integer pageNum,
            @RequestParam(defaultValue = "10") Integer pageSize) {
        return Result.success(new PageResult<>());
    }
    
    @GetMapping("/user/{userId}/history")
    public Result<List<Order>> getOrderHistory(@PathVariable Long userId, @RequestParam(defaultValue = "10") Integer limit) {
        return Result.success(List.of());
    }
    
    @PostMapping("/calculate")
    public Result<OrderCalculationVO> calculateOrder(@Valid @RequestBody CalculateOrderReq req) {
        return Result.success(new OrderCalculationVO());
    }
    
    @PostMapping("/{id}/apply-discount")
    public Result<Order> applyDiscount(@PathVariable Long id, @RequestParam String discountCode) {
        return Result.success(new Order());
    }
    
    @DeleteMapping("/{id}/discount")
    public Result<Order> removeDiscount(@PathVariable Long id) {
        return Result.success(new Order());
    }
    
    @GetMapping("/{id}/items")
    public Result<List<OrderItemVO>> getOrderItems(@PathVariable Long id) {
        return Result.success(List.of());
    }
    
    @PostMapping("/{id}/items")
    public Result<OrderItemVO> addOrderItem(@PathVariable Long id, @Valid @RequestBody AddOrderItemReq req) {
        return Result.success(new OrderItemVO());
    }
    
    @DeleteMapping("/{id}/items/{itemId}")
    public Result<Void> removeOrderItem(@PathVariable Long id, @PathVariable Long itemId) {
        return Result.success(null);
    }
    
    @PutMapping("/{id}/items/{itemId}")
    public Result<OrderItemVO> updateOrderItem(@PathVariable Long id, @PathVariable Long itemId, @RequestParam Integer quantity) {
        return Result.success(new OrderItemVO());
    }
    
    @GetMapping("/{id}/payment")
    public Result<PaymentInfoVO> getPaymentInfo(@PathVariable Long id) {
        return Result.success(new PaymentInfoVO());
    }
    
    @GetMapping("/{id}/shipping")
    public Result<ShippingInfoVO> getShippingInfo(@PathVariable Long id) {
        return Result.success(new ShippingInfoVO());
    }
    
    @PutMapping("/{id}/shipping-address")
    public Result<Order> updateShippingAddress(@PathVariable Long id, @Valid @RequestBody UpdateShippingAddressReq req) {
        return Result.success(new Order());
    }
    
    @PostMapping("/{id}/confirm")
    public Result<Order> confirmOrder(@PathVariable Long id) {
        return Result.success(new Order());
    }
    
    @PostMapping("/{id}/complete")
    public Result<Order> completeOrder(@PathVariable Long id) {
        return Result.success(new Order());
    }
    
    @PostMapping("/{id}/refund")
    public Result<OrderRefundVO> requestRefund(@PathVariable Long id, @Valid @RequestBody OrderRefundReq req) {
        return Result.success(new OrderRefundVO());
    }
    
    @GetMapping("/{id}/refund")
    public Result<OrderRefundVO> getRefundInfo(@PathVariable Long id) {
        return Result.success(new OrderRefundVO());
    }
    
    @GetMapping("/statistics")
    public Result<OrderStatisticsVO> getStatistics(
            @RequestParam(required = false) String startDate,
            @RequestParam(required = false) String endDate) {
        return Result.success(new OrderStatisticsVO());
    }
    
    @GetMapping("/statistics/user/{userId}")
    public Result<UserOrderStatisticsVO> getUserStatistics(@PathVariable Long userId) {
        return Result.success(new UserOrderStatisticsVO());
    }
    
    @GetMapping("/{id}/invoice")
    public Result<InvoiceVO> getInvoice(@PathVariable Long id) {
        return Result.success(new InvoiceVO());
    }
    
    @PostMapping("/{id}/invoice")
    public Result<InvoiceVO> createInvoice(@PathVariable Long id, @Valid @RequestBody CreateInvoiceReq req) {
        return Result.success(new InvoiceVO());
    }
    
    @GetMapping("/export")
    public Result<String> exportOrders(
            @RequestParam(required = false) String startDate,
            @RequestParam(required = false) String endDate,
            @RequestParam(required = false) Integer status) {
        return Result.success("export_file_url");
    }
    
    @PostMapping("/batch-cancel")
    public Result<Void> batchCancel(@RequestBody List<Long> ids, @RequestParam(required = false) String reason) {
        return Result.success(null);
    }
    
    @GetMapping("/pending-count")
    public Result<Long> getPendingCount() {
        return Result.success(0L);
    }
    
    @GetMapping("/user/{userId}/pending-count")
    public Result<Long> getUserPendingCount(@PathVariable Long userId) {
        return Result.success(0L);
    }
    
    @PostMapping("/{id}/remark")
    public Result<Order> addRemark(@PathVariable Long id, @RequestParam String remark) {
        return Result.success(new Order());
    }
    
    @GetMapping("/{id}/logs")
    public Result<List<OrderLogVO>> getOrderLogs(@PathVariable Long id) {
        return Result.success(List.of());
    }
}

class CreateOrderReq {
    private Long userId;
    private List<OrderItemReq> items;
    private String shippingAddress;
    private String receiverName;
    private String receiverPhone;
    private String paymentMethod;
    private String discountCode;
    private String remark;
    
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    public List<OrderItemReq> getItems() { return items; }
    public void setItems(List<OrderItemReq> items) { this.items = items; }
    public String getShippingAddress() { return shippingAddress; }
    public void setShippingAddress(String shippingAddress) { this.shippingAddress = shippingAddress; }
    public String getReceiverName() { return receiverName; }
    public void setReceiverName(String receiverName) { this.receiverName = receiverName; }
    public String getReceiverPhone() { return receiverPhone; }
    public void setReceiverPhone(String receiverPhone) { this.receiverPhone = receiverPhone; }
    public String getPaymentMethod() { return paymentMethod; }
    public void setPaymentMethod(String paymentMethod) { this.paymentMethod = paymentMethod; }
    public String getDiscountCode() { return discountCode; }
    public void setDiscountCode(String discountCode) { this.discountCode = discountCode; }
    public String getRemark() { return remark; }
    public void setRemark(String remark) { this.remark = remark; }
}

class OrderItemReq {
    private Long productId;
    private Integer quantity;
    private String spec;
    
    public Long getProductId() { return productId; }
    public void setProductId(Long productId) { this.productId = productId; }
    public Integer getQuantity() { return quantity; }
    public void setQuantity(Integer quantity) { this.quantity = quantity; }
    public String getSpec() { return spec; }
    public void setSpec(String spec) { this.spec = spec; }
}

class UpdateOrderReq {
    private String shippingAddress;
    private String receiverName;
    private String receiverPhone;
    private String remark;
    
    public String getShippingAddress() { return shippingAddress; }
    public void setShippingAddress(String shippingAddress) { this.shippingAddress = shippingAddress; }
    public String getReceiverName() { return receiverName; }
    public void setReceiverName(String receiverName) { this.receiverName = receiverName; }
    public String getReceiverPhone() { return receiverPhone; }
    public void setReceiverPhone(String receiverPhone) { this.receiverPhone = receiverPhone; }
    public String getRemark() { return remark; }
    public void setRemark(String remark) { this.remark = remark; }
}

class OrderTrackVO {
    private String orderNo;
    private Integer status;
    private String carrier;
    private String trackingNo;
    private List<TrackPointVO> trackPoints;
    
    public String getOrderNo() { return orderNo; }
    public void setOrderNo(String orderNo) { this.orderNo = orderNo; }
    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }
    public String getCarrier() { return carrier; }
    public void setCarrier(String carrier) { this.carrier = carrier; }
    public String getTrackingNo() { return trackingNo; }
    public void setTrackingNo(String trackingNo) { this.trackingNo = trackingNo; }
    public List<TrackPointVO> getTrackPoints() { return trackPoints; }
    public void setTrackPoints(List<TrackPointVO> trackPoints) { this.trackPoints = trackPoints; }
}

class TrackPointVO {
    private String time;
    private String location;
    private String description;
    
    public String getTime() { return time; }
    public void setTime(String time) { this.time = time; }
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
}

class OrderCalculationVO {
    private BigDecimal subtotal;
    private BigDecimal discount;
    private BigDecimal shipping;
    private BigDecimal tax;
    private BigDecimal total;
    
    public BigDecimal getSubtotal() { return subtotal; }
    public void setSubtotal(BigDecimal subtotal) { this.subtotal = subtotal; }
    public BigDecimal getDiscount() { return discount; }
    public void setDiscount(BigDecimal discount) { this.discount = discount; }
    public BigDecimal getShipping() { return shipping; }
    public void setShipping(BigDecimal shipping) { this.shipping = shipping; }
    public BigDecimal getTax() { return tax; }
    public void setTax(BigDecimal tax) { this.tax = tax; }
    public BigDecimal getTotal() { return total; }
    public void setTotal(BigDecimal total) { this.total = total; }
}

class CalculateOrderReq {
    private List<OrderItemReq> items;
    private String discountCode;
    private String shippingAddress;
    
    public List<OrderItemReq> getItems() { return items; }
    public void setItems(List<OrderItemReq> items) { this.items = items; }
    public String getDiscountCode() { return discountCode; }
    public void setDiscountCode(String discountCode) { this.discountCode = discountCode; }
    public String getShippingAddress() { return shippingAddress; }
    public void setShippingAddress(String shippingAddress) { this.shippingAddress = shippingAddress; }
}

class OrderItemVO {
    private Long id;
    private Long productId;
    private String productName;
    private String productImage;
    private BigDecimal price;
    private Integer quantity;
    private String spec;
    private BigDecimal subtotal;
    
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getProductId() { return productId; }
    public void setProductId(Long productId) { this.productId = productId; }
    public String getProductName() { return productName; }
    public void setProductName(String productName) { this.productName = productName; }
    public String getProductImage() { return productImage; }
    public void setProductImage(String productImage) { this.productImage = productImage; }
    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }
    public Integer getQuantity() { return quantity; }
    public void setQuantity(Integer quantity) { this.quantity = quantity; }
    public String getSpec() { return spec; }
    public void setSpec(String spec) { this.spec = spec; }
    public BigDecimal getSubtotal() { return subtotal; }
    public void setSubtotal(BigDecimal subtotal) { this.subtotal = subtotal; }
}

class AddOrderItemReq {
    private Long productId;
    private Integer quantity;
    private String spec;
    
    public Long getProductId() { return productId; }
    public void setProductId(Long productId) { this.productId = productId; }
    public Integer getQuantity() { return quantity; }
    public void setQuantity(Integer quantity) { this.quantity = quantity; }
    public String getSpec() { return spec; }
    public void setSpec(String spec) { this.spec = spec; }
}

class PaymentInfoVO {
    private String paymentNo;
    private BigDecimal amount;
    private String paymentMethod;
    private Integer status;
    private String paidAt;
    
    public String getPaymentNo() { return paymentNo; }
    public void setPaymentNo(String paymentNo) { this.paymentNo = paymentNo; }
    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }
    public String getPaymentMethod() { return paymentMethod; }
    public void setPaymentMethod(String paymentMethod) { this.paymentMethod = paymentMethod; }
    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }
    public String getPaidAt() { return paidAt; }
    public void setPaidAt(String paidAt) { this.paidAt = paidAt; }
}

class ShippingInfoVO {
    private String shippingNo;
    private String carrier;
    private String trackingNo;
    private Integer status;
    private String shippedAt;
    private String deliveredAt;
    
    public String getShippingNo() { return shippingNo; }
    public void setShippingNo(String shippingNo) { this.shippingNo = shippingNo; }
    public String getCarrier() { return carrier; }
    public void setCarrier(String carrier) { this.carrier = carrier; }
    public String getTrackingNo() { return trackingNo; }
    public void setTrackingNo(String trackingNo) { this.trackingNo = trackingNo; }
    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }
    public String getShippedAt() { return shippedAt; }
    public void setShippedAt(String shippedAt) { this.shippedAt = shippedAt; }
    public String getDeliveredAt() { return deliveredAt; }
    public void setDeliveredAt(String deliveredAt) { this.deliveredAt = deliveredAt; }
}

class UpdateShippingAddressReq {
    private String shippingAddress;
    private String receiverName;
    private String receiverPhone;
    
    public String getShippingAddress() { return shippingAddress; }
    public void setShippingAddress(String shippingAddress) { this.shippingAddress = shippingAddress; }
    public String getReceiverName() { return receiverName; }
    public void setReceiverName(String receiverName) { this.receiverName = receiverName; }
    public String getReceiverPhone() { return receiverPhone; }
    public void setReceiverPhone(String receiverPhone) { this.receiverPhone = receiverPhone; }
}

class UserOrderStatisticsVO {
    private Long totalOrders;
    private Long pendingOrders;
    private Long completedOrders;
    private BigDecimal totalSpent;
    
    public Long getTotalOrders() { return totalOrders; }
    public void setTotalOrders(Long totalOrders) { this.totalOrders = totalOrders; }
    public Long getPendingOrders() { return pendingOrders; }
    public void setPendingOrders(Long pendingOrders) { this.pendingOrders = pendingOrders; }
    public Long getCompletedOrders() { return completedOrders; }
    public void setCompletedOrders(Long completedOrders) { this.completedOrders = completedOrders; }
    public BigDecimal getTotalSpent() { return totalSpent; }
    public void setTotalSpent(BigDecimal totalSpent) { this.totalSpent = totalSpent; }
}

class InvoiceVO {
    private Long id;
    private String invoiceNo;
    private String title;
    private String content;
    private BigDecimal amount;
    private String issuedAt;
    
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getInvoiceNo() { return invoiceNo; }
    public void setInvoiceNo(String invoiceNo) { this.invoiceNo = invoiceNo; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }
    public String getIssuedAt() { return issuedAt; }
    public void setIssuedAt(String issuedAt) { this.issuedAt = issuedAt; }
}

class CreateInvoiceReq {
    private String title;
    private String content;
    private String email;
    
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
}

class OrderLogVO {
    private Long id;
    private String action;
    private String description;
    private String operator;
    private String createdAt;
    
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getAction() { return action; }
    public void setAction(String action) { this.action = action; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getOperator() { return operator; }
    public void setOperator(String operator) { this.operator = operator; }
    public String getCreatedAt() { return createdAt; }
    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }
}

File: en/kotlin-springmvc/src/main/kotlin/com/example/controller/OrderController.kt
package com.example.controller

import com.example.dto.PageResult
import com.example.dto.Result
import com.example.model.Order
import org.springframework.web.bind.annotation.*
import java.math.BigDecimal

@RestController
@RequestMapping("/api/orders")
class OrderController {
    
    @GetMapping
    fun listOrders(
        @RequestParam(required = false) userId: Long?,
        @RequestParam(required = false) orderNo: String?,
        @RequestParam(required = false) status: Int?,
        @RequestParam(required = false) startDate: String?,
        @RequestParam(required = false) endDate: String?,
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int
    ): Result<PageResult<Order>> {
        val result = PageResult<Order>()
        return Result.success(result)
    }
    
    @PostMapping
    fun createOrder(@RequestBody req: CreateOrderReq): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @GetMapping("/{id}")
    fun getOrder(@PathVariable id: Long): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @PutMapping("/{id}")
    fun updateOrder(@PathVariable id: Long, @RequestBody req: UpdateOrderReq): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @DeleteMapping("/{id}")
    fun deleteOrder(@PathVariable id: Long): Result<Void> {
        return Result.success(null)
    }
    
    @PostMapping("/{id}/cancel")
    fun cancelOrder(@PathVariable id: Long, @RequestParam(required = false) reason: String?): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @PutMapping("/{id}/status")
    fun updateStatus(@PathVariable id: Long, @RequestParam status: Int): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @GetMapping("/{id}/track")
    fun trackOrder(@PathVariable id: Long): Result<OrderTrackVO> {
        val track = OrderTrackVO()
        return Result.success(track)
    }
    
    @GetMapping("/user/{userId}")
    fun getUserOrders(
        @PathVariable userId: Long,
        @RequestParam(required = false) status: Int?,
        @RequestParam(defaultValue = "1") pageNum: Int,
        @RequestParam(defaultValue = "10") pageSize: Int
    ): Result<PageResult<Order>> {
        val result = PageResult<Order>()
        return Result.success(result)
    }
    
    @GetMapping("/user/{userId}/history")
    fun getOrderHistory(@PathVariable userId: Long, @RequestParam(defaultValue = "10") limit: Int): Result<List<Order>> {
        val orders = listOf<Order>()
        return Result.success(orders)
    }
    
    @PostMapping("/calculate")
    fun calculateOrder(@RequestBody req: CalculateOrderReq): Result<OrderCalculationVO> {
        val calculation = OrderCalculationVO()
        return Result.success(calculation)
    }
    
    @PostMapping("/{id}/apply-discount")
    fun applyDiscount(@PathVariable id: Long, @RequestParam discountCode: String): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @DeleteMapping("/{id}/discount")
    fun removeDiscount(@PathVariable id: Long): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @GetMapping("/{id}/items")
    fun getOrderItems(@PathVariable id: Long): Result<List<OrderItemVO>> {
        val items = listOf<OrderItemVO>()
        return Result.success(items)
    }
    
    @PostMapping("/{id}/items")
    fun addOrderItem(@PathVariable id: Long, @RequestBody req: AddOrderItemReq): Result<OrderItemVO> {
        val item = OrderItemVO()
        return Result.success(item)
    }
    
    @DeleteMapping("/{id}/items/{itemId}")
    fun removeOrderItem(@PathVariable id: Long, @PathVariable itemId: Long): Result<Void> {
        return Result.success(null)
    }
    
    @PutMapping("/{id}/items/{itemId}")
    fun updateOrderItem(
        @PathVariable id: Long,
        @PathVariable itemId: Long,
        @RequestParam quantity: Int
    ): Result<OrderItemVO> {
        val item = OrderItemVO()
        return Result.success(item)
    }
    
    @GetMapping("/{id}/payment")
    fun getPaymentInfo(@PathVariable id: Long): Result<PaymentInfoVO> {
        val payment = PaymentInfoVO()
        return Result.success(payment)
    }
    
    @GetMapping("/{id}/shipping")
    fun getShippingInfo(@PathVariable id: Long): Result<ShippingInfoVO> {
        val shipping = ShippingInfoVO()
        return Result.success(shipping)
    }
    
    @PutMapping("/{id}/shipping-address")
    fun updateShippingAddress(@PathVariable id: Long, @RequestBody req: UpdateShippingAddressReq): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @PostMapping("/{id}/confirm")
    fun confirmOrder(@PathVariable id: Long): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @PostMapping("/{id}/complete")
    fun completeOrder(@PathVariable id: Long): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @PostMapping("/{id}/refund")
    fun requestRefund(@PathVariable id: Long, @RequestBody req: OrderRefundReq): Result<OrderRefundVO> {
        val refund = OrderRefundVO()
        return Result.success(refund)
    }
    
    @GetMapping("/{id}/refund")
    fun getRefundInfo(@PathVariable id: Long): Result<OrderRefundVO> {
        val refund = OrderRefundVO()
        return Result.success(refund)
    }
    
    @GetMapping("/statistics")
    fun getStatistics(
        @RequestParam(required = false) startDate: String?,
        @RequestParam(required = false) endDate: String?
    ): Result<OrderStatisticsVO> {
        val statistics = OrderStatisticsVO()
        return Result.success(statistics)
    }
    
    @GetMapping("/statistics/user/{userId}")
    fun getUserStatistics(@PathVariable userId: Long): Result<UserOrderStatisticsVO> {
        val statistics = UserOrderStatisticsVO()
        return Result.success(statistics)
    }
    
    @GetMapping("/{id}/invoice")
    fun getInvoice(@PathVariable id: Long): Result<InvoiceVO> {
        val invoice = InvoiceVO()
        return Result.success(invoice)
    }
    
    @PostMapping("/{id}/invoice")
    fun createInvoice(@PathVariable id: Long, @RequestBody req: CreateInvoiceReq): Result<InvoiceVO> {
        val invoice = InvoiceVO()
        return Result.success(invoice)
    }
    
    @GetMapping("/export")
    fun exportOrders(
        @RequestParam(required = false) startDate: String?,
        @RequestParam(required = false) endDate: String?,
        @RequestParam(required = false) status: Int?
    ): Result<String> {
        val fileUrl = ""
        return Result.success(fileUrl)
    }
    
    @PostMapping("/batch-cancel")
    fun batchCancel(@RequestBody ids: List<Long>, @RequestParam(required = false) reason: String?): Result<Void> {
        return Result.success(null)
    }
    
    @GetMapping("/pending-count")
    fun getPendingCount(): Result<Long> {
        val count = 0L
        return Result.success(count)
    }
    
    @GetMapping("/user/{userId}/pending-count")
    fun getUserPendingCount(@PathVariable userId: Long): Result<Long> {
        val count = 0L
        return Result.success(count)
    }
    
    @PostMapping("/{id}/remark")
    fun addRemark(@PathVariable id: Long, @RequestParam remark: String): Result<Order> {
        val order = Order()
        return Result.success(order)
    }
    
    @GetMapping("/{id}/logs")
    fun getOrderLogs(@PathVariable id: Long): Result<List<OrderLogVO>> {
        val logs = listOf<OrderLogVO>()
        return Result.success(logs)
    }
}

data class CreateOrderReq(
    val userId: Long?,
    val items: List<OrderItemReq>?,
    val shippingAddress: String?,
    val receiverName: String?,
    val receiverPhone: String?,
    val paymentMethod: String?,
    val discountCode: String?,
    val remark: String?
)

data class OrderItemReq(
    val productId: Long?,
    val quantity: Int?,
    val spec: String?
)

data class UpdateOrderReq(
    val shippingAddress: String?,
    val receiverName: String?,
    val receiverPhone: String?,
    val remark: String?
)

data class OrderTrackVO(
    val orderNo: String? = null,
    val status: Int? = null,
    val carrier: String? = null,
    val trackingNo: String? = null,
    val trackPoints: List<TrackPointVO>? = null
)

data class TrackPointVO(
    val time: String? = null,
    val location: String? = null,
    val description: String? = null
)

data class OrderCalculationVO(
    val subtotal: BigDecimal? = null,
    val discount: BigDecimal? = null,
    val shipping: BigDecimal? = null,
    val tax: BigDecimal? = null,
    val total: BigDecimal? = null
)

data class CalculateOrderReq(
    val items: List<OrderItemReq>?,
    val discountCode: String?,
    val shippingAddress: String?
)

data class OrderItemVO(
    val id: Long? = null,
    val productId: Long? = null,
    val productName: String? = null,
    val productImage: String? = null,
    val price: BigDecimal? = null,
    val quantity: Int? = null,
    val spec: String? = null,
    val subtotal: BigDecimal? = null
)

data class AddOrderItemReq(
    val productId: Long?,
    val quantity: Int?,
    val spec: String?
)

data class PaymentInfoVO(
    val paymentNo: String? = null,
    val amount: BigDecimal? = null,
    val paymentMethod: String? = null,
    val status: Int? = null,
    val paidAt: String? = null
)

data class ShippingInfoVO(
    val shippingNo: String? = null,
    val carrier: String? = null,
    val trackingNo: String? = null,
    val status: Int? = null,
    val shippedAt: String? = null,
    val deliveredAt: String? = null
)

data class UpdateShippingAddressReq(
    val shippingAddress: String?,
    val receiverName: String?,
    val receiverPhone: String?
)

data class OrderRefundVO(
    val id: Long? = null,
    val orderId: Long? = null,
    val amount: BigDecimal? = null,
    val reason: String? = null,
    val status: Int? = null,
    val processedAt: String? = null
)

data class OrderRefundReq(
    val amount: BigDecimal?,
    val reason: String?
)

data class OrderStatisticsVO(
    val totalOrders: Long? = null,
    val pendingOrders: Long? = null,
    val completedOrders: Long? = null,
    val cancelledOrders: Long? = null,
    val totalRevenue: BigDecimal? = null,
    val averageOrderValue: BigDecimal? = null
)

data class UserOrderStatisticsVO(
    val totalOrders: Long? = null,
    val pendingOrders: Long? = null,
    val completedOrders: Long? = null,
    val totalSpent: BigDecimal? = null
)

data class InvoiceVO(
    val id: Long? = null,
    val invoiceNo: String? = null,
    val title: String? = null,
    val content: String? = null,
    val amount: BigDecimal? = null,
    val issuedAt: String? = null
)

data class CreateInvoiceReq(
    val title: String?,
    val content: String?,
    val email: String?
)

data class OrderLogVO(
    val id: Long? = null,
    val action: String? = null,
    val description: String? = null,
    val operator: String? = null,
    val createdAt: String? = null
)
